lib/pure/math.nim

```diff
@@ -949,18 +949,25 @@ func frexp*[T: float32|float64](x: T): tuple[frac: T, exp: int] {.inline.} =
     doAssert frexp(8.0) == (0.5, 4)
     doAssert frexp(-8.0) == (-0.5, 4)
     doAssert frexp(0.0) == (0.0, 0)
+
     # special cases:
-    when not defined(windows):
-      doAssert frexp(-0.0) == (-0.0, 0) # signbit preserved for +-0
-      doAssert frexp(Inf).frac == Inf # +- Inf preserved
-      doAssert frexp(NaN).frac.isNaN
+    doAssert frexp(-0.0) == (-0.0, 0) # signbit preserved for +-0
```

Comment:
This does not actually check for exactly -0.0 (since `0.0 == -0.0`), however, it is good for illustration (and it's not wrong), so I kept it.

Reply:
since then we've added signbit; use it here